src/librustc_lexer/src/lib.rs

```diff
@@ -692,27 +630,34 @@ impl Cursor<'_> {
         false
     }
 
-    /// Eats the double-quoted string and returns an `UnvalidatedRawStr`.
-    fn raw_double_quoted_string(&mut self, prefix_len: usize) -> UnvalidatedRawStr {
+    /// Eats the double-quoted string and returns `n_hashes` and an error if encountered.
+    fn raw_double_quoted_string(&mut self, prefix_len: usize) -> (u16, Option<RawStrError>) {
+        // Wrap the actual function to handle the error with too many hashes.
+        // This way, it eats the whole raw string.
+        let (n_hashes, err) = self.raw_string_unvalidated(prefix_len);
+        // Only up to 65535 `#`s are allowed in raw strings
+        match u16::try_from(n_hashes) {
+            Ok(num) => (num, err),
+            //FIXME Feedback request: Is it ok to lie about num_hashes in the error case?
```

Comment:
Totally ok.
You could use `u16::MAX` to lie less, but that doesn't really matter because it always produces an error and never happens in practice anyway.

Reply:
I thought about that, but decided agains it, because it would probably be easier for a user to realise that something is buggy, if rustc ignores the hashes entirely.